stdlib/contrib/bonitoo-io/zenoss/zenoss_test.go

```diff
@@ -0,0 +1,282 @@
+    summary: r.description,
+    device: r.node,
+    component: "CPU",
+    severity: r.severity,
+    eventClass: "/App",
+    eventClassKey: "",
+    message: "",
+    collector: "localhost",
+}))
+
+csv.from(csv:data) |> process()
+`)
+
+	if err != nil {
+		t.Error(err)
+	}
+	_ = scope
+}
+
+func TestZenossPost(t *testing.T) {
+	s := NewServer(t)
+	defer s.Close()
+
+	testCases := []struct {
+		name     string
+		URL      string
+		addEvent AddEvent
+		fn       string
+	}{
+		{
+			name: "alert with defaults",
+			URL:  s.URL,
+			addEvent: AddEvent{
+				Action: "EventsRouter",
+				Method: "add_event",
+				Data: []Event{
+					{
+						Summary:    "some alert",
+						EventClass: "/App",
+						Severity:   "Warning",
+					},
+				},
+				Type: "rpc",
+				TID:  1,
+			},
+			fn: "zenoss.endpoint(url: url, username: username, password: password)",
+		},
+		{
+			name: "alert with all fields",
+			URL:  s.URL,
+			addEvent: AddEvent{
+				Action: "CustomRouter",
+				Method: "new_event",
+				Data: []Event{
+					{
+						Summary:    "CPU-2 is too busy",
+						Device:     "10.1.2.3",
+						Component:  "CPU",
+						EventClass: "/App",
+						Severity:   "Warning",
+						Collector:  "localhost",
+					},
+				},
+				Type: "doc",
+				TID:  1,
+			},
+			fn: "zenoss.endpoint(url: url, username: username, password: password, action: action, method: method, type: type, tid: tid)",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			s.Reset()
+
+			event := tc.addEvent.Data[0]
+			fluxString := `import "csv"
+import "contrib/bonitoo-io/zenoss"
+
+url = "` + tc.URL + `"
+username = "admin"
+password = "12345"
+action = "` + tc.addEvent.Action + `"
+method = "` + tc.addEvent.Method + `"
+type = "` + tc.addEvent.Type + `"
+tid = ` + strconv.Itoa(tc.addEvent.TID) + `
+
+data = "
+#group,false,false,false,false,false,false,false,false,false
+#datatype,string,string,string,string,string,string,string,string,string
+#default,_result,,,,,,,,
+,result,table,node,metric_type,resource,metric_name,message_key,description,severity
+,,0,10.1.1.1,CPU,CPU-1,usage_idle,Alert-#1001,CPU-1 too busy,Critical
+,,0,` + strings.Join([]string{event.Device, event.Component, "CPU-2", "usage_user", "Alert-#1002", event.Summary, event.Severity}, ",") + `
+"
+
+endpoint = ` + tc.fn + `(mapFn: (r) => ({
+    summary: r.description,
+    device: r.node,
+    component: r.metric_type,
+    severity: r.severity,
+    eventClass: "/App",
+    eventClassKey: "",
+    message: "",
+    collector: "` + event.Collector + `",
+}))
+
+csv.from(csv:data) |> endpoint()`
+
+			prog, err := lang.Compile(fluxString, runtime.Default, time.Now())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ctx := flux.NewDefaultDependencies().Inject(context.Background())
+			query, err := prog.Start(ctx, &memory.Allocator{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			res := <-query.Results()
+			_ = res
```

Comment:
Can you explain what's going on here? If you need to retrieve the result and discard it, you can just do:

```go
<-query.Results()
```

It might be also good to add a timeout to this.

```go
t := time.NewTimer(200 * time.Millisecond)
select {
case <-t.C:
    t.Fatal("timeout while waiting for query results")
case <-query.Results():
    t.Stop()
}
```

Reply:
Thank you! It was all copy&paste from a test in existing package (`slack` I think) which I took as a paradigm.